dts/bindings/i2c/st,stm32-i2c-v1.yaml

```diff
@@ -13,3 +13,8 @@ properties:
 
     interrupts:
       required: true
+
```

Comment:
I don't believe this belongs in devicetree: it's not a hardware description issue, but an application policy.  If it's to be configurable it should be via Kconfig.  If necessary I can expedite the work on a generic Kconfig option for this.

Reply:
I am fine if we move it to Kconfig, The best option is timeout to be configurable per transaction (e.g. add a new parameter to read / write API). As a workaround we could configured it via Kconfig per controller.
As mentioned before, I wasn't expecting this PR to be merged. I am only keeping this PR open as a workaround for someone who needs it until the asynchronous APIs is ready and merged.